steps/25_split_package/step.md

```diff
@@ -0,0 +1,148 @@
+release process. The intention is to improve maintainability and scalability, while
+allowing for more flexible development and release cycles.
+
+## Contents
+
+1. Problem statement
+2. Description of proposed solution
+3. Motivation
+4. Discussion and comparison of alternative solutions
+5. Detailed description of design and implementation
+    - Package structure
+    - Migration strategy
+    - Interoperability and testing
+    - Release strategy
+    - Documentation and user support
+6. Potential concerns and areas to monitor
+
+## Problem statement
+
+The current monolithic structure of sktime presents several challenges:
+
+- Rapid growth in codebase size and complexity.
+- Increasing difficulty in managing and resolving dependency conflicts, especially with
+  soft dependencies.
+- Release process bottlenecked by a single release manager and the need to synchronise
+  all modules.
+- Inconsistent dependency bound management and insufficient testing of all documented
+  bounds.
+- Limited and non-exhaustive interoperability testing across modules, particularly for
+  estimators with soft dependencies.
+
+## Description of proposed solution
+
+This proposal suggests to split sktime into a set of independent, interoperable
+packages, each corresponding to a major module (e.g., forecasting, classification,
+transformation). A lightweight `sktime-core` package will provide shared base classes
+and utilities, ensuring interoperability and a unified interface. Pipeline and
+composition logic will initially reside in `sktime-core`, with the option to split into
+a dedicated `sktime-pipeline` package if complexity warrants. Each component package
+will manage its own dependencies, extras, and release cadence, following semantic
+versioning. A clear migration path and enhanced interoperability testing will be
+established.
+
+## Motivation
+
+- **Maintainability:** Smaller, focused packages are easier to maintain, test, and
+  document.
+- **Scalability:** Modularisation allows the project to scale with new contributions and
+  features.
+- **Dependency management:** Isolating dependencies per package reduces conflicts and
+  installation issues.
+- **Release flexibility:** Independent release cycles enable faster bug fixes and
+  feature delivery.
+- **User experience:** Users can install only what they need, reducing bloat and
+  complexity.
+
+## Discussion and comparison of alternative solutions
+
+- **Status quo:** Retaining the monolithic structure avoid the overhead of managing
+  multiple packages, but would continue to exacerbate maintenance and dependency issues
+  as the project grows.
+- **Partial modularisation:** Splitting only some modules would not fully address
+  dependency and release bottlenecks and coordination challenges.
+- **Meta-package (`sktime-all`):** While useful for transition, maintaining a
+  meta-package long-term increases maintenance overhead and can reintroduce dependency
+  conflicts.
+
```

Comment:
there is another alternative: improving the sub-packaging usint `scikit-base` - already estimators have their own dependency sets.

That would leave a single pypi package, but might mitigate a significant part of the issues.
See the registry module, `python_dependencies` tag, `tests:vm` tag, and `deps` or `craft`.